examples/platform/silabs/SoftwareFaultReports.cpp

```diff
@@ -89,18 +89,18 @@ void OnSoftwareFaultEventHandler(const char * faultRecordString)
 extern "C" __attribute__((used)) void debugHardfault(uint32_t * sp)
 {
 #if SILABS_LOG_ENABLED
-    uint32_t cfsr  = SCB->CFSR;
-    uint32_t hfsr  = SCB->HFSR;
-    uint32_t mmfar = SCB->MMFAR;
-    uint32_t bfar  = SCB->BFAR;
-    uint32_t r0    = sp[0];
-    uint32_t r1    = sp[1];
-    uint32_t r2    = sp[2];
-    uint32_t r3    = sp[3];
-    uint32_t r12   = sp[4];
-    uint32_t lr    = sp[5];
-    uint32_t pc    = sp[6];
-    uint32_t psr   = sp[7];
+    [[maybe_unused]] uint32_t cfsr  = SCB->CFSR;
```

Comment:
Given that this is guarded by an ifdef, why would these be unused? Is there a case where SILABS_LOG_ENABLED is on, but the chip log functions are noops?

Reply:
if we disable only error logging

Reply:
Added a check on chip_logging in .gni files to ensure turning all logging off removes dead code

Reply:
OH. gotcha.